src/main/kotlin/org/rust/cargo/runconfig/command/RunCargoCommandActionBase.kt

```diff
@@ -35,10 +37,19 @@ abstract class RunCargoCommandActionBase(icon: Icon) : AnAction(icon) {
             ?.let { cargoProjects.findProjectForFile(it) }
             ?.let { return it }
 
-        return cargoProjects.allProjects.firstOrNull()
+        val cargoPanel = ToolWindowManager.getInstance(e.project!!)
```

Comment:
IntelliJ platform has `DataContext` interface to pass some data into actions.
Usually you just pass some key into `AnActionEvent.getData()` and get the corresponding data.
To provide data you need to implement `DataProvider` interface.

In our case, `CargoToolWindowPanel` already implements `DataProvider` as an inheritor of `SimpleToolWindowPanel`. Moreover, it provides a selected project for `DetachCargoProjectAction` via `DetachCargoProjectAction.CARGO_PROJECT_TO_DETACH` key.
Let's generalize this api to use it in this case too.

I think we should to the following:
* move out `DetachCargoProjectAction.CARGO_PROJECT_TO_DETACH` key from `DetachCargoProjectAction`, for example, into `CargoToolWindow.kt` file
* rename it to `SELECTED_CARGO_PROJECT` and use `"Cargo.selectedProject"` as key name (or use other informative names)
* use it in `RunCargoCommandActionBase` to get selected cargo project